### PR TITLE
[Feature] Add Cosmos2 i2v pipeline

### PR DESCRIPTION
**Pull request by kevin314**

### CODE REVIEW

File: fastvideo/configs/models/vaes/cosmosvae.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import torch
+
+from fastvideo.configs.models.vaes.base import VAEArchConfig, VAEConfig
+
+
+@dataclass
+class CosmosVAEArchConfig(VAEArchConfig):
+    _class_name: str = "AutoencoderKLWan"
+    _diffusers_version: str = "0.34.0.dev0"
+    _name_or_path: str = ""
```
Comment (SolitaryThinker) [inline fastvideo/configs/models/vaes/cosmosvae.py]: These fields should be pop/removed in the loader so can be removed. You can refer to how wan's vae config is defined

File: fastvideo/configs/models/vaes/cosmosvae.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    arch_config: CosmosVAEArchConfig = field(default_factory=CosmosVAEArchConfig)
+    use_feature_cache: bool = True
+
+    use_tiling: bool = False
+    use_temporal_tiling: bool = False
+    use_parallel_tiling: bool = False
+
+    def __post_init__(self):
+        self.blend_num_frames = (self.tile_sample_min_num_frames -
+                                 self.tile_sample_stride_num_frames) * 2
```
Comment (SolitaryThinker) [inline fastvideo/configs/models/vaes/cosmosvae.py]: Newline char

File: fastvideo/configs/pipelines/cosmos.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        print(f"Hidden state device: {hidden_state.device}")
+        # Replace NaN values with zeros to avoid pipeline failure
+        hidden_state = hidden_state.masked_fill(torch.isnan(hidden_state), 0.0)
+    
+    # Return raw last_hidden_state (no truncation/padding)
+    return hidden_state
+
+
+@dataclass
+class CosmosVideoConfigFixed(CosmosVideoConfig):
```
Comment (SolitaryThinker) [inline fastvideo/configs/pipelines/cosmos.py]: Why is this needed?

File: fastvideo/configs/pipelines/cosmos.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    """Fixed Cosmos Video Config that matches original Cosmos2 Video2World configuration."""
+    
+    def update_model_arch(self, config: dict) -> None:
```
Comment (SolitaryThinker) [inline fastvideo/configs/pipelines/cosmos.py]: Did you align against diffusers or original cosmos?

File: fastvideo/configs/sample/cosmos.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+from dataclasses import dataclass, field
+
+from fastvideo.configs.sample.base import CacheParams
+
+
+@dataclass
+class CosmosTeaCacheParams(CacheParams):
```
Comment (SolitaryThinker) [inline fastvideo/configs/sample/cosmos.py]: This can be removed for now, but you should add a default for cosmos sampling params. Refer to wan

File: fastvideo/layers/layernorm.py [hunk 6]
```diff
@@ -39,6 +39,22 @@ def __init__(
         if self.has_weight:
             self.weight = nn.Parameter(self.weight)
 
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
```
Comment (SolitaryThinker) [inline fastvideo/layers/layernorm.py]: can you change this to forward_diffusers?

File: fastvideo/layers/rotary_embedding.py [hunk 7]
```diff
@@ -47,6 +47,59 @@ def _rotate_gptj(x: torch.Tensor) -> torch.Tensor:
     return x.flatten(-2)
 
 
+def apply_rotary_emb(
```
Comment (SolitaryThinker) [inline fastvideo/layers/rotary_embedding.py:50]: Maybe we should consider making this cosmos2 specific?

File: fastvideo/layers/visual_embedding.py [hunk 8]
```diff
@@ -177,3 +177,79 @@ def unpatchify(x, t, h, w, patch_size, channels) -> torch.Tensor:
     imgs = x.reshape(shape=(x.shape[0], c, t * pt, h * ph, w * pw))
 
     return imgs
+
+
+def get_timestep_embedding(
```
Comment (SolitaryThinker) [inline fastvideo/layers/visual_embedding.py:182]: Maybe also make the architecture specific

File: fastvideo/models/encoders/t5.py [hunk 9]
```diff
@@ -297,10 +299,12 @@ def forward(
     ) -> torch.Tensor:
         bs, seq_len, _ = hidden_states.shape
         num_seqs = bs
-        n, c = self.n_heads, self.d_model // self.total_num_heads
+        #n, c = self.n_heads, self.d_model // self.total_num_heads
```
Comment (SolitaryThinker) [inline fastvideo/models/encoders/t5.py]: delete this

File: fastvideo/models/encoders/t5.py [hunk 10]
```diff
@@ -297,10 +299,12 @@ def forward(
...
+        n, c = self.n_heads, self.key_value_proj_dim
         qkv, _ = self.qkv_proj(hidden_states)
         # Projection of 'own' hidden state (self-attention). No GQA here.
-        q, k, v = qkv.split(self.inner_dim, dim=-1)
+        #q, k, v = qkv.split(self.inner_dim, dim=-1)
```
Comment (SolitaryThinker) [inline fastvideo/models/encoders/t5.py]: delete this

File: fastvideo/models/schedulers/scheduling_flow_match_euler_discrete.py [hunk 11]
```diff
@@ -336,9 +348,9 @@ def set_timesteps(
         sigmas_array: np.ndarray
         if sigmas is None:
             if timesteps_array is None:
-                timesteps_array = np.linspace(self._sigma_to_t(self.sigma_max),
-                                              self._sigma_to_t(self.sigma_min),
-                                              num_inference_steps)
+                t_max = self._sigma_to_t(self.sigma_max)
```
Comment (SolitaryThinker) [inline fastvideo/models/schedulers/scheduling_flow_match_euler_discrete.py:351]: maybe make a separate version of this scheduler

File: fastvideo/models/registry.py [hunk 12]
```diff
@@ -239,7 +241,6 @@ def register_model(
 
     def _raise_for_unsupported(self, architectures: list[str]) -> NoReturn:
         all_supported_archs = self.get_supported_archs()
-
```
Comment (SolitaryThinker) [inline fastvideo/models/registry.py:242]: revert this

File: fastvideo/pipelines/stages/denoising.py
Comment (SolitaryThinker) [inline fastvideo/pipelines/stages/denoising.py:1]: we should separate the different versions of denoising stage into their own files

File: fastvideo/pipelines/stages/input_validation.py [hunk 13]
```diff
@@ -41,7 +41,7 @@ def _generate_seeds(self, batch: ForwardBatch,
         batch.seeds = seeds
         # Peiyuan: using GPU seed will cause A100 and H100 to generate different results...
         batch.generator = [
-            torch.Generator("cpu").manual_seed(seed) for seed in seeds
+            torch.Generator(device="cpu").manual_seed(seed) for seed in seeds
```
Comment (SolitaryThinker) [inline fastvideo/pipelines/stages/input_validation.py]: revert please

File: fastvideo/pipelines/stages/latent_preparation.py
Comment (SolitaryThinker) [inline fastvideo/pipelines/stages/latent_preparation.py:1]: maybe also separate these

File: fastvideo/pipelines/stages/utils.py
Comment (SolitaryThinker) [inline fastvideo/pipelines/stages/utils.py:1]: make arch specific

File: fastvideo/tests/encoders/test_t5_encoder.py
Comment (SolitaryThinker) [inline fastvideo/tests/encoders/test_t5_encoder.py:1]: test both

File: fastvideo/worker/multiproc_executor.py [hunk 14]
```diff
@@ -32,6 +32,13 @@ def _init_executor(self) -> None:
         self.world_size = self.fastvideo_args.num_gpus
         self.shutting_down = False
 
+        # Initialize CUDA before setting up multiprocessing to ensure
```
Comment (SolitaryThinker) [inline fastvideo/worker/multiproc_executor.py]: remove
